common/src/main/java/com/graphicsfuzz/common/util/FragmentShaderJobToVertexShaderJob.java

```diff
@@ -0,0 +1,226 @@
+import com.graphicsfuzz.common.ast.visitors.StandardVisitor;
+import com.graphicsfuzz.common.glslversion.ShadingLanguageVersion;
+import com.graphicsfuzz.common.transformreduce.GlslShaderJob;
+import com.graphicsfuzz.common.transformreduce.ShaderJob;
+import com.graphicsfuzz.common.typing.ScopeTrackingVisitor;
+import com.graphicsfuzz.util.ArgsUtil;
+import com.graphicsfuzz.util.Constants;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+
+public final class FragmentShaderJobToVertexShaderJob {
+
+  private FragmentShaderJobToVertexShaderJob() {
+    // Utility class
+  }
+
+  private static Optional<Integer> maybeGetArrayCount(VariableDeclInfo vdi) {
+    if (vdi.hasArrayInfo()) {
+      return Optional.of(vdi.getArrayInfo().getConstantSize());
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Creates a shader job with the given translation unit as its fragment shader, a pipeline state
+   * that provides a random value for every uniform declared in the shader, and if needed a vertex
+   * shader that provides outputs for the fragment shader's inputs.
+   *
+   * @param tu A fragment shader.
+   * @return A shader job that includes the fragment shader.
+   */
+
+
+  public static ShaderJob convertShaderJob(ShaderJob sj) throws Exception {
+    final PipelineInfo pipelineInfo = sj.getPipelineInfo();
+    final List<TranslationUnit> shaders = new ArrayList<>();
+    
+    // Get the current fragment shader and change it to vertex shader
+    TranslationUnit vertexShader = sj.getFragmentShader().get();
+    vertexShader.setShaderKind(ShaderKind.VERTEX);
+
+    // Create pass-through fragment shader
+    final TranslationUnit fragmentShader = new TranslationUnit(ShaderKind.FRAGMENT,
+        Optional.of(ShadingLanguageVersion.GLSL_430), Arrays.asList(
+        new PrecisionDeclaration("precision highp float;"),
+        new VariablesDeclaration(new QualifiedType(BasicType.VEC4,
+            Arrays.asList(new LayoutQualifierSequence(
+                    new LocationLayoutQualifier(0)),
+                TypeQualifier.SHADER_INPUT)),
+            new VariableDeclInfo(Constants.GLF_COLOR, null, null)),
+        new VariablesDeclaration(new QualifiedType(BasicType.VEC4,
+            Arrays.asList(new LayoutQualifierSequence(
+                    new LocationLayoutQualifier(0)),
+                TypeQualifier.SHADER_OUTPUT)),
+            new VariableDeclInfo("frag_color", null, null)),
+        new FunctionDefinition(
+            new FunctionPrototype("main", VoidType.VOID, Collections.emptyList()),
+            new BlockStmt(Arrays.asList(
```

Comment:
Nit: use Collections.singletonList(...) for a 1-element list.

Reply:
This is one bit where I'd rather leave it as is, to remain similar to other code like it and to make it easier to expand when it inevitably needs to.